pkg/services/validator/aws.go

```diff
@@ -57,44 +68,421 @@ func readAwsPlugin(vc *components.ValidatorConfig, k8sClient kubernetes.Interfac
+	specified in the passed in policy documents.
+	`)
+
+	validatePolicies, err := prompts.ReadBool("Enable Iam Policy validation", true)
+	if err != nil {
+		return err
+	}
+	if !validatePolicies {
+		c.Validator.IamPolicyRules = nil
+		return nil
+	}
+	for i, r := range c.Validator.IamPolicyRules {
+		r := r
+		if err := readIamPolicyRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+	addRules := true
+	if c.Validator.IamPolicyRules == nil {
+		c.Validator.IamPolicyRules = make([]vpawsapi.IamPolicyRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another policy rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readIamPolicyRule(c, nil, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another policy rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+func readIamPolicyRule(c *components.AWSPluginConfig, r *vpawsapi.IamPolicyRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpawsapi.IamPolicyRule{
+			IamPolicyARN: "",
+			Policies:     []vpawsapi.PolicyDocument{},
+		}
+	}
+	err := initAwsRule(r, "iam policy", ruleNames)
+	if err != nil {
+		return err
+	}
+	if r.IamPolicyARN == "" {
+		policyArn, err := prompts.ReadTextRegex("IAM Policy ARN", "", "Invalid Policy ARN", cfg.PolicyArnRegex)
+		if err != nil {
+			return err
+		}
+		r.IamPolicyARN = policyArn
+	}
+
+	addPolicies := true
+	for addPolicies {
+		policyDoc, err := readIamPolicy()
+		if err != nil {
+			return err
+		}
+
+		r.Policies = append(r.Policies, policyDoc)
+		addPolicies, err = prompts.ReadBool("Add another policy document", false)
+		if err != nil {
+			return err
+		}
+	}
+	if idx == -1 {
+		c.Validator.IamPolicyRules = append(c.Validator.IamPolicyRules, *r)
+	} else {
+		c.Validator.IamPolicyRules[idx] = *r
+	}
 	return nil
 }
 
+func readIamPolicy() (vpawsapi.PolicyDocument, error) {
+	policyDoc := vpawsapi.PolicyDocument{}
+	inputType, err := prompts.Select("Add policy document via", []string{"Local Filepath", "File Editor"})
+	if err != nil {
+		return policyDoc, err
+	}
+
+	var policyBytes []byte
+	if inputType == "Local Filepath" {
+		policyFile, err := prompts.ReadFilePath("Policy Document Filepath", "", "Invalid policy document path", false)
+		if err != nil {
+			return policyDoc, err
+		}
+
+		file, err := os.Open(policyFile) //#nosec
+		if err != nil {
+			return policyDoc, err
+		}
+		defer file.Close()
+
+		policyBytes, err = os.ReadFile(policyFile) //#nosec
+		if err != nil {
+			return policyDoc, err
+		}
+	} else {
+		log.InfoCLI("Configure Policy Document")
+		time.Sleep(2 * time.Second)
+		policyFile, err := prompts.EditFileValidatedByFullContent(cfg.AWSPolicyDocumentPrompt, "", prompts.ValidateJson, 1)
+		if err != nil {
+			return policyDoc, err
+		}
+		policyBytes = []byte(policyFile)
+	}
+
+	var policy awspolicy.Policy
+	err = policy.UnmarshalJSON(policyBytes)
```

Comment:
It would be nice if the file selection + unmarshaling was more forgiving. I.e., allow the user to try selecting a different file if they somehow provide one that's invalid.